source3/smbd/trans2.c

```diff
@@ -229,8 +242,8 @@ NTSTATUS get_ea_value(TALLOC_CTX *mem_ctx,
 				ea_name, val, attr_size);
 	}
 
-	if (sizeret == -1 && errno == ERANGE && attr_size != 65536) {
-		attr_size = 65536;
+	if (sizeret == -1 && errno == ERANGE && attr_size <= max_xattr_size) {
+		attr_size = sizeret;
 		goto again;
```

Comment:
Do I miss something or `sizeret = -1` when we enter the condition, hence `attr_size = -1` and we are going to allocate that? I guess whole code here had something like `attr_size *= 2;` in mind.